### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by emiliazm**
Hello!
### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements
- Create a method to decode a Morse code character, takes a string parameter, and return the corresponding character in uppercase (e.g. `decode_char(".-")` returns `"A"`).
- Create a method to decode an entire word in Morse code, takes a string parameter, and return the string representation. Every character in a word will be separated by a single space (e.g. `decode_word("-- -.--")` returns `"MY"`).
- Create a method to decode the entire message in Morse code, takes a string parameter, and return the string representation. Every word will be separated by 3 spaces (e.g.
    ```
      decode("-- -.--   -. .- -- .")
    ```
  
  returns `"MY NAME"`).
- Reuse methods whenever possible. Make sure that your code is DRY.
- Once your code is ready use it to decode the message from the old bottle:

```
      .-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ...
```

### CODE REVIEW

File: decode.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    '.--' => 'W', '-..-' => 'X',
+    '-.--' => 'Y', '--..' => 'Z'
+  }
+
+  decoder[str]
+end
+
+def decode_str(str)
+  words_arr = str.split('  ')
+  phrase = ''
```
Comment (Gambit142) [inline decode.rb]: - [x] Kindly create a ruby method that takes a **_morse code input word_** delimited by a **_single space_** as a parameter. You should split through input word as the next step. The method should iterate through the morse code input word and use the method `decode_char` to translate every letter. 👍 

#### Hint: `decode_word('-- -.--')` should return `"MY"`

image

File: decode.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  words_arr.each do |morse_word|
+    morse_word_arr = morse_word.split
+    morse_word_arr.each do |morse_char|
+      phrase += decode_char(morse_char)
+    end
+    phrase += ' '
+  end
+  print phrase
```
Comment (Gambit142) [inline decode.rb]: - [x] This method should take a **_morse code input sentence_** as a parameter. You should split the morse code input sentence by **_three spaces_**. After that, you should iterate through the sentence and use the method that you will create for splitting through input words to concatenate the sentence. 

#### Hint:

`decode("-- -.-- -. .- -- .")` should return `"MY NAME"`.

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @emiliazm @Trictonicmp 👋, 

Good job so far!

## Highlights
- Linters checks passed 💯 
- Use of correct Git flow 💯 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Trictonicmp** [comment]
Hello @Gambit142 ! Thank you for your feedback, we made the changes required.

**Review by DammyShittu — APPROVED** [review]
Hi team,

## APPROVED 🟢

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**